api/domain/ordersDomain.js

```diff
@@ -120,49 +124,68 @@ exports.changeQuantity = async (order, products) => {
 }
 
 exports.changeCurrency = async (prevOrder, updatedOrder) => {
-    for (let n = 0; n < updatedOrder.products.length; n++) {
-        await OrderHelpers.substractQuantityAndTotalPrice(prevOrder, updatedOrder.products[n].id)
-        
+    const productsToUpdate = updatedOrder.products
+
+    await this.deleteProductsPrevCurrency(prevOrder, updatedOrder, productsToUpdate)
+    await this.addProductsNewCurrency(updatedOrder, productsToUpdate)
+}
+
+exports.deleteProductsPrevCurrency = async (prevOrder, updatedOrder, productsToUpdate) => {
```

Comment:
maybe you can pass here updatedOrder.dataValues in line 129 instead of just updatedOrder. Then you can simplify line 141 and 142

Reply:
And I think exports.deleteProductsOldCurrency would be better

Reply:
I cannot pass .dataValues because sequelize is operating on updatedOrder (for this case) and not on updatedOrder.dataValues